### PR TITLE
community: chat_message_history.dynamodb Update DynamoDB to optionally use an item per message

### PR DESCRIPTION
**Pull request by LeeroyHannigan**
Thank you for contributing to LangChain!
- **Description:**
     This add an additional parameter to `DynamoDBChatMessageHistory` that allows each message to be contained in its own item. This is a best practice for DynamoDB, and avoids hitting the 400KB item size limit. 
- **Issue:**
    The issue is the chance of hitting the 400KB items size limit in DynamoDB. `history_size` works to some degree, but at the risk of losing relevant parts of the conversation.
- **Dependencies:** any dependencies required for this change
    Imported `boto3.dynamodb.conditions import Key` which is already a dependency with `boto3`
- **Twitter handle:** if your PR gets announced, and you'd like a mention, we'll gladly shout you out!
    `leeroy_hannigan`

- [ ] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [ ] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

File: libs/community/langchain_community/chat_message_histories/dynamodb.py [hunk 1]
```diff
@@ -62,6 +62,8 @@ def __init__(
         ttl_key_name: str = "expireAt",
         history_size: Optional[int] = None,
         history_messages_key: Optional[str] = "History",
+        *,
+        vertical_partition: Optional[bool] = None,
```
Comment (baskaryan) [inline libs/community/langchain_community/chat_message_histories/dynamodb.py:66]: could we documentation for this param?

**Comment by LeeroyHannigan** [comment]
@ccurme @efriis I believe we have 6 upvotes for this PR now.

**Comment by efriis** [comment]
hey @LeeroyHannigan ! @baskaryan left a review that wasn't acted upon on september 2nd - could you address that?

**Comment by efriis** [comment]
Also CI is failing - could you get that passing? Here's how you can run it locally:

First, [install poetry](https://python-poetry.org/docs/#installing-with-the-official-installer), e.g. with `curl -sSL https://install.python-poetry.org | python3 -` on non-windows machines (see link for other ways of installing)

Then, for example if you're in community, you can run

```bash
# enter package you're checking, e.g. community:
cd libs/community

# format and lint
poetry install --with lint,typing --sync
make format lint

# test
poetry install --with lint,typing,test --sync
make test

# extended tests - install extra deps with uv - not a requirement but it's how I do it
poetry install --with lint,typing,test --sync
poetry run pip install uv
poetry run uv pip install -r extended_testing_deps.txt
make extended_tests

# run a specific test - helpful if trying to just fix a single test
poetry run pytest --disable-socket --allow-unix-socket tests/unit_tests -k test_my_test_name
```

**Comment by efriis** [comment]
closing as stale - feel free to reopen if still interesting, also might be worth filing an issue to migrate this in the https://github.com/langchain-ai/langchain-aws repo

**Pull request closed without merging**